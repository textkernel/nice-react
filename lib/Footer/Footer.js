import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

const Link = ({ children, href, target, className, onClick }) => (<li>
    <a href={ href } target={ target } className={ className } onClick={ onClick }>
        { children }
    </a>
</li>);

Link.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.node,
        PropTypes.array
    ]),
    target: PropTypes.string,
    className: PropTypes.string,
    onClick: PropTypes.func
};
Link.defaultProps = {
    children: null,
    href: '#',
    target: null,
    className: null,
    onClick: null
};

class Footer extends Component {
    static propTypes = {
        children: PropTypes.oneOfType([
            PropTypes.node,
            PropTypes.array
        ]),
        copyright: PropTypes.bool,
        fluid: PropTypes.bool
    }

    static defaultProps = {
        children: null,
        copyright: true,
        fluid: false
    }

    renderCopyright() {
        const { copyright } = this.props;

        if (!copyright) {
            return null;
        }

        return (<p className="pull-left">
            &copy; { (new Date()).getFullYear() }
            <span className="text-brand">
                textkernel
            </span>
        </p>);
    }

    renderMenu() {
        const { children } = this.props;

        if (!children) {
            return null;
        }

        return (<ul className="nice-footer-menu">
            { children }
        </ul>);
    }

    render() {
        const { copyright, fluid } = this.props;

        return (
            <footer className="nice-footer">
                <div className={ classNames({
                    'nice-container': !fluid,
                    'nice-container-fluid': fluid
                }) }>
                    { this.renderCopyright() }
                    { this.renderMenu() }
                </div>
            </footer>
        );
    }
}

Footer.Link = Link;

export default Footer;
