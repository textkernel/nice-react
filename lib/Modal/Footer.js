import React from 'react';
import PropTypes from 'prop-types';

const ModalFooter = ({ children }) => {
    if (!children) {
        return null;
    }

    return (
        <div className="nice-modal-footer">
            { children }
        </div>
    );
};

ModalFooter.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.node,
        PropTypes.string,
        PropTypes.number
    ])
};

ModalFooter.defaultProps = {
    children: null
};

export default ModalFooter;
