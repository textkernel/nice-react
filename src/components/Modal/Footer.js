import React from 'react';
import classnames from 'classnames';
import PropTypes from 'prop-types';

const Footer = ({ children, className, style }) => {
    if (!children) {
        return null;
    }

    return (
        <div className={classnames('nice-modal-footer', className)} style={style}>
            { children }
        </div>
    );
};

Footer.propTypes = {
    className: PropTypes.string,
    children: PropTypes.oneOfType([
        PropTypes.node,
        PropTypes.string,
        PropTypes.number
    ]),
    style: PropTypes.object
};

Footer.defaultProps = {
    className: null,
    children: null,
    style: {}
};

export default Footer;
