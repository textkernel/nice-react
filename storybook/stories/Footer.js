import React from 'react';
import { storiesOf } from '@storybook/react';
import { withKnobsOptions, withKnobs, boolean } from '@storybook/addon-knobs';
import { withInfo } from '@storybook/addon-info';

import { Footer } from 'nice-react';

const stories = storiesOf('Footer', module);

stories.addDecorator((story, context) => withInfo('')(story)(context));
stories.addDecorator(withKnobs);

stories.add('Default', () => {
    const fluid = boolean('Fluid layout', false);
    const copyright = boolean('Include copyright', true);

    return (<Footer copyright={ copyright } fluid={ fluid }>
        <Footer.Link className="icon fa-balance-scale" href="https://github.com/textkernel/nice/blob/master/LICENSE" target="_blank">
            Project License
        </Footer.Link>
        <Footer.Link className="icon fa-github" href="https://github.com/textkernel/nice/" target="_blank">
            Source
        </Footer.Link>
        <Footer.Link className="additional-icon fw frontify" href="https://app.frontify.com/d/apB5ernOJZzj/textkernel-branding-guideline" target="_blank">
            Brand Guidelines
        </Footer.Link>
        <Footer.Link className="product-icon textkernel" href="http://textkernel.com" target="_blank">
            Textkernel
        </Footer.Link>
    </Footer>)
});
