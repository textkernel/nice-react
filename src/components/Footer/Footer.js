import React, { Component } from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';
import './Footer.scss';

const Link = ({ children, href, target, className, onClick }) => (<li>
    <a href={ href } target={ target } className={ className } onClick={ onClick }>
        { children }
    </a>
</li>);

Link.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.node,
        PropTypes.array
    ]),
    href: PropTypes.string,
    target: PropTypes.string,
    className: PropTypes.string,
    onClick: PropTypes.func
};

Link.defaultProps = {
    children: null,
    href: '#',
    target: null,
    className: null,
    onClick: null
};

class Footer extends Component {
    renderCopyright() {
        const { copyright } = this.props;

        if (!copyright) {
            return null;
        }

        return (<p className="pull-left">
            &copy; { (new Date()).getFullYear() }
            <span className="text-brand">
                textkernel
            </span>
        </p>);
    }

    renderMenu() {
        const { children } = this.props;

        if (!children) {
            return null;
        }

        return (<ul className="nice-footer-menu">
            { children }
        </ul>);
    }

    render() {
        const { className, fluid, style } = this.props;

        return (
            <footer
                className={ classNames('nice-footer', className) }
                style={ style }
            >
                <div className={ classNames({
                    'nice-container': !fluid,
                    'nice-container-fluid': fluid
                }) }>
                    { this.renderCopyright() }
                    { this.renderMenu() }
                </div>
            </footer>
        );
    }
}

Footer.Link = Link;

Footer.propTypes = {
    children: PropTypes.oneOfType([
        PropTypes.node,
        PropTypes.array
    ]),
    className: PropTypes.string,
    copyright: PropTypes.bool,
    fluid: PropTypes.bool,
    style: PropTypes.object
};

Footer.defaultProps = {
    children: null,
    className: '',
    copyright: true,
    fluid: false,
    style: null
};

export default Footer;
