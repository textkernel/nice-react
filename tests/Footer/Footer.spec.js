import React from 'react';
import Footer from '../../lib/Footer';

describe('<Footer />', () => {
    it('should render a default Footer', () => {
        const wrapper = shallow(<Footer copyright>
            <Footer.Link className="icon fa-balance-scale" href="https://github.com/textkernel/nice/blob/master/LICENSE" target="_blank">
                Project License
            </Footer.Link>
            <Footer.Link className="icon fa-github" href="https://github.com/textkernel/nice/" target="_blank">
                Source
            </Footer.Link>
            <Footer.Link className="additional-icon fw frontify" href="https://app.frontify.com/d/apB5ernOJZzj/textkernel-branding-guideline" target="_blank">
                Brand Guidelines
            </Footer.Link>
            <Footer.Link className="product-icon textkernel" href="http://textkernel.com" target="_blank">
                Textkernel
            </Footer.Link>
        </Footer>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a fluid Footer without copyright', () => {
        const wrapper = shallow(<Footer fluid>
            <Footer.Link>
                Test
            </Footer.Link>
        </Footer>);
        expect(wrapper).toMatchSnapshot();
    });

    it('should render a Footer with custom classes / styles', () => {
        const wrapper = shallow(<Footer className="test" style={{ display: 'none' }}>
            <Footer.Link>
                Test
            </Footer.Link>
        </Footer>);
        expect(wrapper).toMatchSnapshot();
    });
});
