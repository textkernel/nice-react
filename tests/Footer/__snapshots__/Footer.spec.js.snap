// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render a Footer with custom classes / styles 1`] = `
<footer
  className="nice-footer test"
  style={
    Object {
      "display": "none",
    }
  }
>
  <div
    className="nice-container"
  >
    <p
      className="pull-left"
    >
      © 
      2018
      <span
        className="text-brand"
      >
        textkernel
      </span>
    </p>
    <ul
      className="nice-footer-menu"
    >
      <Link
        className={null}
        href="#"
        onClick={null}
        target={null}
      >
        Test
      </Link>
    </ul>
  </div>
</footer>
`;

exports[`<Footer /> should render a default Footer 1`] = `
<footer
  className="nice-footer"
  style={null}
>
  <div
    className="nice-container"
  >
    <p
      className="pull-left"
    >
      © 
      2018
      <span
        className="text-brand"
      >
        textkernel
      </span>
    </p>
    <ul
      className="nice-footer-menu"
    >
      <Link
        className="icon fa-balance-scale"
        href="https://github.com/textkernel/nice/blob/master/LICENSE"
        onClick={null}
        target="_blank"
      >
        Project License
      </Link>
      <Link
        className="icon fa-github"
        href="https://github.com/textkernel/nice/"
        onClick={null}
        target="_blank"
      >
        Source
      </Link>
      <Link
        className="additional-icon fw frontify"
        href="https://app.frontify.com/d/apB5ernOJZzj/textkernel-branding-guideline"
        onClick={null}
        target="_blank"
      >
        Brand Guidelines
      </Link>
      <Link
        className="product-icon textkernel"
        href="http://textkernel.com"
        onClick={null}
        target="_blank"
      >
        Textkernel
      </Link>
    </ul>
  </div>
</footer>
`;

exports[`<Footer /> should render a fluid Footer without copyright 1`] = `
<footer
  className="nice-footer"
  style={null}
>
  <div
    className="nice-container-fluid"
  >
    <p
      className="pull-left"
    >
      © 
      2018
      <span
        className="text-brand"
      >
        textkernel
      </span>
    </p>
    <ul
      className="nice-footer-menu"
    >
      <Link
        className={null}
        href="#"
        onClick={null}
        target={null}
      >
        Test
      </Link>
    </ul>
  </div>
</footer>
`;
